cmd/scollector/collectors/continuous.go

```diff
@@ -0,0 +1,95 @@
+package collectors
+
+import (
+	"reflect"
+	"regexp"
+	"runtime"
+	"time"
+
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+type ContinuousCollector struct {
+	F    func(collectorStatsChan chan<- *ContinuousCollectorStats)
+	name string
+	tags opentsdb.TagSet
+	init func()
+}
+
+type ContinuousCollectorStats struct {
+	duration float64
+	result   int
+}
+
+var ContinuousCollectorVars struct {
+	dpChan                chan<- *opentsdb.DataPoint
+	quit                  <-chan struct{}
+	reTwoOrMoreUnderscore *regexp.Regexp
+	nameWorkerChan        map[string]chan *ContinuousCollectorStats
+}
+
+func init() {
+	ContinuousCollectorVars.reTwoOrMoreUnderscore = regexp.MustCompile("[_]{2,}")
+	ContinuousCollectorVars.nameWorkerChan = make(map[string]chan *ContinuousCollectorStats)
+}
+
+func (c *ContinuousCollector) Init() {
+	if c.init != nil {
+		c.init()
+	}
+}
+
+func (c *ContinuousCollector) Run(dpChan chan<- *opentsdb.DataPoint, quit <-chan struct{}) {
```

Comment:
Help me understand why continuous collector is needed. It looks like you are still running each query on an interval. Is it because you maintain a set of connections? `database/sql` should maintain good enough connection pooling that I wouldn't worry about that. 

Why won't interval collector work for this?